README.md

```diff
@@ -8,12 +8,47 @@ The goal of [Accessors.jl](https://github.com/JuliaObjects/Accessors.jl) is to m
 It is the successor of [Setfield.jl](https://github.com/jw3126/Setfield.jl).
 
 # Usage
-Updating immutable data was never easier:
+Say you have some immutable data structure, such as a `NamedTuple`:
 ```julia
-using Accessors
-@set obj.a.b.c = d
+julia> nt = (a=1, b=2)
+(a = 1, b = 2)
+
+```
+If you try something like `nt.b=3`, it will throw an error. But 
+using Accessors, we can change it anyways:
+```julia
+julia> using Accessors
+
+julia> @set nt.b=3
+(a = 1, b = 3)
+```
+
+Note that this only returns a mutated version of `nt`, and *does not overwrite* the 
+value bound to `nt`:
+```julia
+julia> nt
+(a = 1, b = 2)
+```
+
+To overwrite the old definition, we can rebind `nt` to the mutated version:
+```julia
+julia> nt = @set nt.b=3
+(a = 1, b = 3)
+
+julia> nt
+(a = 1, b = 3)
+```
+
+As this is a common use case, the convenience macro `@reset` overwrites what you are mutating:
```

Comment:
```suggestion
As this is a common use case, the convenience macro `@reset` rebinds the updated value:
```

Reply:
perhaps "rebinds the variable to the updated value"? is a bit more explicit?